### ISSUE TITLE
Prevent element highlighting from highlighting the entire page

### ISSUE THREAD

**Issue by swoh816** [obsidianmd/obsidian-clipper#191]
**Describe the solution you'd like**

Problem: In the current version, you can select the entire document in highlight mode, overriding all the sentence and paragraph highlights. I don't think the document-wise highlight can go, because if I want to highlight the entire document, I can just clip it without highlighting anything.

Solution: Disable document-wise highlight in highlight mode.

**Comment by kepano** [comment]
The extension already does this to some extent. It blocks highlighting the `html` and `body` elements. However some websites have large `div` containers that might contain the entire page. Since we do still want to have element highlighting it's tricky to differentiate between which elements should be allowed and which shouldn't.

Some ideas that come to mind:

- An option to disable element highlighting, for everything except for paragraphs and images
- A modifier key that temporarily disables element highlighting

In the meantime you can also use Cmd+Z to cancel a highlight if you accidentally highlight the background.

**Comment by swoh816** [comment]
Indeed, only after I clicked the document in highlight mode, screaming out I lost all the highlights, I found out I can undo using Cmd+Z and I was relieved.

I agree it's tricky. I like the way it can highlight any element in the document (except the entire document). Potentially, traversing the hierarchy of tags could resolve it. For example, filter the parent nodes that contain children nodes with <p> tags or <img> tags.

**Comment by AriTheElk** [comment]
> I agree it's tricky. I like the way it can highlight any element in the document (except the entire document). Potentially, traversing the hierarchy of tags could resolve it. For example, filter the parent nodes that contain children nodes with `<p>` tags or `<img>` tags.

I think the problem with that approach, is it assumes semantically correct html. In web development, divs are probably the most common element wrapping text 😓

We could potentially look for the first "fork" in the DOM tree, so the first element that has a sibling, and treat that as the highest highlightable element? It wouldn't entirely fix this issue, but would at least prevent entire page highlights

**Comment by qqrs** [comment]
> I found out I can undo using Cmd+Z and I was relieved

I came here with this problem. I like Obsidian Clipper but I was about to give up on it because it was so frustrating to repeatedly lose 20+ highlights on a page by clicking the wrong spot.

It's great to find out there is a way to undo. That is enough of a workaround for me.

I think it's very rare someone with a lot of highlights will want to replace them all by highlighting the entire page. Maybe there could be a warning dialog? Or a message reminding them how they can undo?

**Comment by kepano** [comment]
This should be fixed in 0.11.6
